astropy/stats/bayesian_blocks.py

```diff
@@ -0,0 +1,424 @@
+"""
+Bayesian Blocks for Time Series Analysis
+========================================
+
+Dynamic programming algorithm for solving a piecewise-constant model for
+various datasets. This is based on the algorithm presented in Scargle
+et al 2012 [1]_. This code was ported from the astroML project [2]_.
+
+Applications include:
+
+- finding an optimial histogram with adaptive bin widths
+- finding optimal segmentation of time series data
+- detecting inflection points in the rate of event data
+
+The primary interface to these routines is the :func:`bayesian_blocks`
+function. This module provides fitness functions suitable for three types
+of data:
+
+- Irregularly-spaced event data via the :class:`Events` class
+- Regularly-spaced event data via the :class:`RegularEvents` class
+- Irregularly-spaced point measurements via the :class:`PointMeasures` class
+
+For more fine-tuned control over the fitness functions used, it is possible
+to define custom :class:`FitnessFunc` classes directly and use them with
+the :func:`bayesian_blocks` routine.
+
+
+One common application of the Bayesian Blocks algorithm is the determination
+of optimal adaptive-width histogram bins. This uses the same fitness function
+as for irregularly-spaced time series events. The easiest interface for
+creating Bayesian Blocks histograms is the :func:``astropy.stats.histogram``
+function.
+
+References
+----------
+.. [1] http://adsabs.harvard.edu/abs/2012arXiv1207.5578S
+.. [2] http://astroML.org/ http://github.com/astroML/astroML/
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+
+# TODO: implement other fitness functions from appendix B of Scargle 2012
+
+__all__ = ['FitnessFunc', 'Events', 'RegularEvents', 'PointMeasures',
+           'bayesian_blocks']
+
+
+def bayesian_blocks(t, x=None, sigma=None,
+                    fitness='events', **kwargs):
+    """Bayesian Blocks Implementation
+
+    This is a flexible implementation of the Bayesian Blocks algorithm
+    described in Scargle 2012 [1]_
+
+    Parameters
+    ----------
+    t : array_like
+        data times (one dimensional, length N)
+    x : array_like (optional)
+        data values
+    sigma : array_like or float (optional)
+        data errors
+    fitness : str or object
+        the fitness function to use for the model.
+        If a string, the following options are supported:
+
+        - 'events' : binned or unbinned event data
+            extra arguments are `p0`, which gives the false alarm probability
+            to compute the prior, or `gamma` which gives the slope of the
```

Comment:
This should be double back-ticks too - basically single back-ticks are reserved for references that actually resolve to functions or methods.